src/cim/components/modals/AddHostModal.tsx

```diff
@@ -0,0 +1,307 @@
+                    name={`macMapping[${index}].macAddress`}
+                    // eslint-disable-next-line
+                    inputError={(error as any)?.[index]?.macAddress}
+                  />
+                </GridItem>
+                <GridItem span={5}>
+                  <InputField
+                    name={`macMapping[${index}].name`}
+                    // eslint-disable-next-line
+                    inputError={(error as any)?.[index]?.name}
+                  />
+                </GridItem>
+                {index !== 0 && (
+                  <GridItem span={2}>
+                    <MinusCircleIcon onClick={() => remove(index)} />
+                  </GridItem>
+                )}
+              </React.Fragment>
+            ))}
+            <GridItem span={6}>
+              <Button
+                icon={<PlusCircleIcon />}
+                onClick={() => push({ macAddress: '', name: '' })}
+                variant="link"
+                isInline
+              >
+                Add more
+              </Button>
+            </GridItem>
+          </Grid>
+        )}
+      />
+    </FormGroup>
+  );
+};
+
+const getNMState = (values: AddBmcValues, infraEnv: InfraEnvK8sResource) => {
+  // eslint-disable-next-line
+  const config: any = yaml.load(values.nmState);
+  const nmState = {
+    apiVersion: 'agent-install.openshift.io/v1beta1',
+    kind: 'NMStateConfig',
+    metadata: {
+      generateName: `${infraEnv.metadata?.name}-`,
+      namespace: infraEnv.metadata?.namespace,
+      labels: {
+        infraEnv: infraEnv.metadata?.name,
+        bmcMAC: values.bmcAddress,
+      },
+    },
+    spec: {
+      config,
+      interfaces: values.macMapping.filter((m) => m.macAddress.length && m.name.length),
+    },
+  };
+  return nmState;
+};
+
+const validationSchema = Yup.object({
+  hostname: hostnameValidationSchema.required(),
+  bmcAddress: ipOrDomainValidationSchema.required(),
+  username: Yup.string().required(),
+  password: Yup.string().required(),
+  bootMACAddress: macAddressValidationSchema.required(),
+  nmState: Yup.string().required(),
+  macMapping: Yup.array().of(
+    Yup.object().shape({
+      macAddress: macAddressValidationSchema.required('MAC Address is a required field'),
+      name: Yup.string().required('NIC is a required field'),
+    }),
+  ),
+});
+
+const emptyValues: AddBmcValues = {
+  hostname: '',
+  bmcAddress: '',
+  username: '',
+  password: '',
+  bootMACAddress: '',
+  disableCertificateVerification: true, // TODO(mlibra)
+  online: false, // TODO(mlibra)
+  nmState: '',
+  macMapping: [{ macAddress: '', name: '' }],
+};
+
+type BMCFormProps = {
+  onClose: VoidFunction;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  onCreate: (values: AddBmcValues, nmState?: any) => Promise<any>;
+  hasDHCP: boolean;
+  infraEnv: InfraEnvK8sResource;
+};
+
+const BMCForm: React.FC<BMCFormProps> = ({ onCreate, onClose, hasDHCP, infraEnv }) => {
```

Comment:
nit: What about splitting that file per components?

Reply:
done